shared/string.js

```diff
@@ -0,0 +1,27 @@
+String.prototype.isEmpty = function() {
+  return this.trim() === '';
+};
+
+String.prototype.isBlankOrGitOrBit = function() {
+  return /^$|github.com|bitbucket.org/g.test(this);
+};
+
+String.prototype.toLowerHyphenated = function() {
+  return this.toLowerCase().trim().replace(/\s/g, '-');
+};
+
+String.prototype.toCapitalizeUnhyphenated = function() {
+  return this.replace(/-/g, ' ')
+    .replace(/\b\w/g, function(m) {
+      return m.toUpperCase();
+    })
+    .replace(/\s/g, '');
+};
+
+String.prototype.toUpperCaseUnhyphenated = function() {
```

Comment:
Following your logic we are going to end up with functions like: 

“String”.nonEmptyAndCapitalizeStringRemoveDashAndIsValidFormated();

Where by nature functions should do one thing and should be composable like: 

“String”.trim().nonEmpty().removeDash().ValidFormat().

Let me know if you have more questions: 

> On Apr 21, 2016, at 2:58 PM, Adam Fenton <notifications@github.com> wrote:
> 
> In shared/string.js <https://github.com/moxie-leean/generators/pull/10#discussion_r60645476>:
> 
> > +  return /^$|github.com|bitbucket.org/g.test(this);
> > +};
> > +
> > +String.prototype.toLowerHyphenated = function() {
> > +  return this.toLowerCase().trim().replace(/\s/g, '-');
> > +};
> > +
> > +String.prototype.toCapitalizeUnhyphenated = function() {
> > +  return this.replace(/-/g, ' ')
> > +    .replace(/\b\w/g, function(m) {
> > +      return m.toUpperCase();
> > +    })
> > +    .replace(/\s/g, '');
> > +};
> > +
> > +String.prototype.toUpperCaseUnhyphenated = function() {
> Same here.
> Also why wrap just this.replace(/\W+/g, '') in a function? It's pretty clear what it does by reading the regex
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/moxie-leean/generators/pull/10/files/05edd5d1f72a4f14401c09c9b3c2a119785548c8#r60645476>